module/Company/src/Company/Service/Company.php

```diff
@@ -215,6 +250,24 @@ public function insertCategory($lang, $id, $cat = null)
         $result = $this->getCategoryMapper()->insert($lang, $id, $cat);
         return $result;
     }
+
+    /**
+     * Inserts a label, and binds it to the given package
+     *
+     * @param mixed $packageID
+     */
+    public function insertLabel($lang, $id, $cat = null)
+    {
+        if (!$this->isAllowed('insert')) {
+            $translator = $this->getTranslator();
+            throw new \User\Permissions\NotAllowedException(
+                $translator->translate('You are not allowed to insert a job')
+            );
+        }
+        $result = $this->getLabelMapper()->insert($lang, $id, $cat);
```

Comment:
Just before this line would be the perfect place to create a label model, and send it to the mapper.

Reply:
Yep, did so